### PR TITLE
Login with Redux 🦸🏻‍♂️

### PR DESCRIPTION
**Pull request by moralesbang**
### Summary

- Setup redux to project.
- Add reactotron redux plugin.
- Update authentication logic using redux.

### Trello Card

https://trello.com/c/JCwpekga

### CODE REVIEW

File: src/interfaces/auth.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
+/* eslint-disable prettier/prettier */
```
Comment (marianozicavo) [inline src/interfaces/auth.ts:1]: 👀 why?

Comment (moralesbang) [reply src/interfaces/auth.ts:1]: Sorry. I was flashing :v

File: src/app/screens/Login/index.tsx [hunk 2]
```diff
@@ -38,11 +43,11 @@ function Login() {
...
       />
       <Button
-        buttonStyle={{ ...styles.loginButton, ...(!formIsValid && styles.loginButtonDisabled) }}
+        buttonStyle={{ ...styles.loginButton, ...(submitDisabled && styles.loginButtonDisabled) }}
         buttonTextStyle={styles.loginButtonText}
         onPress={signIn}
-        disabled={!formIsValid}>
-        Sign in
+        disabled={submitDisabled}>
+        {userIsLoading ? 'Loading...' : 'Sign in'}
```
Comment (sfernandez11) [inline src/app/screens/Login/index.tsx:50]: Here you can use the [ActivityIndicator](https://reactnative.dev/docs/activityindicator) for a more dynamic loading animation.
For example, you can pass a prop (`loading`)  to the Button and then:
```
Button() {
...
 return loading ? (
  <ActivityIndicator ... />
  ) : (
  <TouchableOpacity ... />
 );
...
}
```
What do you think?

File: src/config/api.ts [hunk 3]
```diff
@@ -1,12 +1,26 @@
...
 import Reactotron from 'reactotron-react-native';
 
+const deserializer = new CamelcaseSerializer();
+
 const api = create({
   baseURL: Config.API_BASE_URL,
   timeout: 5000
 });
 
+export const authApi = create({
```
Comment (sfernandez11) [inline src/config/api.ts:13]: Why you create two different APIs?

File: src/redux/auth/actions.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
+import AuthService from '@services/AuthService';
+import { UserCredentials } from '@interfaces/auth';
+import { Dispatch } from 'redux';
```
Comment (sfernandez11) [inline src/redux/auth/actions.ts:3]: Same, sort  imports

File: src/config/reactotron.ts [hunk 5]
```diff
@@ -2,20 +2,25 @@ import Reactotron, { asyncStorage, overlay, trackGlobalErrors } from 'reactotron
 import { NativeModules } from 'react-native';
 import tronsauce from 'reactotron-apisauce';
 import { Tron } from '@interfaces/reactotron';
+import { reactotronRedux } from 'reactotron-redux';
```
Comment (sfernandez11) [inline src/config/reactotron.ts:5]: Same

File: src/redux/auth/reducer.ts [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+  userLoading: false,
+  userError: null
+};
+
+const authReducer = (state = initalState, action: AuthActionTypes): AuthState => {
+  switch (action.type) {
+    case actions.LOGIN:
+      return { ...state, userLoading: true };
+    case actions.LOGIN_SUCCESS:
+      return { ...state, user: action.payload, userLoading: false };
```
Comment (sfernandez11) [inline src/redux/auth/reducer.ts:16]: `return { ...state, user: action.payload, userLoading: false, userError: null };` to delete old errors

File: src/app/screens/Login/index.tsx (hunk 2 above)
Comment (moralesbang) [reply src/app/screens/Login/index.tsx:50]: Yeah! Pretty cool. This change has a card that I'm gonna do in the next PR.
https://trello.com/c/sl7sTZV6

File: src/redux/auth/reducer.ts (hunk 6 above)
Comment (moralesbang) [reply src/redux/auth/reducer.ts:16]: I love this comment

File: .eslintrc.js [hunk 7]
```diff
@@ -6,20 +6,44 @@ module.exports = {
...
+          '@app': './src/app',
+          '@assets': './src/app/assets',
+          '@components': './src/app/components',
+          '@screens': './src/app/screens',
+          '@config': './src/config',
+          '@constants': './src/constants',
+          '@interfaces': './src/interfaces',
+          '@redux': './src/redux',
+          '@services': './src/services',
+          '@utils': './src/utils'
```
Comment (sfernandez11) [inline .eslintrc.js:46]: Mmm why this change? The aliases are in `babel.config.js` and `tsconfig.json`, not here 🤔

File: .eslintrc.js [hunk 8]
```diff
@@ -6,20 +6,44 @@ module.exports = {
...
+          },
+          {
+            pattern: 'react-native',
+            group: 'external',
+            position: 'before'
+          }
+        ],
+        pathGroupsExcludedImportTypes: ['react', 'react-native']
+      }
+    ]
```
Comment (sfernandez11) [inline .eslintrc.js:28]: 👏 👏

File: App.tsx [hunk 9]
```diff
@@ -1,8 +1,16 @@
 import '@config/reactotron';
 import React from 'react';
 
+import { Provider } from 'react-redux';
+
 import App from './src/app';
 
+import store from '@redux/store';
```
Comment (sfernandez11) [inline App.tsx:8]: These imports are not correct:
The order is:
1) Global imports (`react`, `react-native`,  `react-redux`, etc...)

Maybe blank line with the new configuration 🤔 

2) Aliases

Blank line

3) Paths with `../...`

Blank line

4) Paths with `./...`

Check this please

File: __tests__/App-test.tsx [hunk 10]
```diff
@@ -4,6 +4,7 @@
 
 import 'react-native';
 import React from 'react';
+
```
Comment (sfernandez11) [inline __tests__/App-test.tsx:7]: I don't like this  blank line between global imports 🤔

File: src/app/components/BackButton/index.tsx [hunk 11]
```diff
@@ -1,9 +1,11 @@
 import React from 'react';
+
```
Comment (sfernandez11) [inline src/app/components/BackButton/index.tsx:2]: Same with all of this case

File: .eslintrc.js (hunk 7 above)
Comment (moralesbang) [reply .eslintrc.js:46]: Sorry. I'm still understanding this. But, you have reason.

File: App.tsx (hunk 9 above)
Comment (moralesbang) [reply App.tsx:8]: Cool

File: src/app/components/TabBarIcon/index.tsx [hunk 12]
```diff
@@ -1,6 +1,7 @@
-import React from 'react';
 import { useRoute } from '@react-navigation/native';
```
Comment (marianozicavo) [inline src/app/components/TabBarIcon/index.tsx:1]: This one should go with the other @

File: src/services/AuthService.ts [hunk 13]
```diff
@@ -0,0 +1,13 @@
+import { ApiResponse } from 'apisauce';
+
+import api from '@config/api';
+import { LoginResponse } from '@interfaces/api';
+import { UserCredentials } from '@interfaces/auth';
+
+type LoginPromise = Promise<ApiResponse<LoginResponse, string>>;
```
Comment (marianozicavo) [inline src/services/AuthService.ts:7]: Is this type required? What happens if you remove it?

File: src/app/index.tsx [hunk 14]
```diff
@@ -1,7 +1,8 @@
 import 'react-native-gesture-handler';
+import { NavigationContainer } from '@react-navigation/native';
```
Comment (marianozicavo) [inline src/app/index.tsx:2]: Same

File: src/app/screens/BookDetail/components/BookInfo/index.tsx [hunk 15]
```diff
@@ -1,17 +1,18 @@
+import { useRoute, RouteProp } from '@react-navigation/native';
```
Comment (marianozicavo) [inline src/app/screens/BookDetail/components/BookInfo/index.tsx:1]: Same
